# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1455696366 28800
#      Wed Feb 17 00:06:06 2016 -0800
# Node ID 41e39c9b35d31854c7e1f1c01453ed723f3a771c
# Parent  7abdce2ae9000170b02e04da2e0a7ca872372981
maint: Clean up makefile fragment generation in initial bootstrap

* bootstrap.conf (bootstrap_post_import_hook): Delete makefile fragments
before generating.  Use ${SHELL} environment variable (with default fallback)
as command interpreter for helper scripts.

diff --git a/bootstrap.conf b/bootstrap.conf
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -182,19 +182,22 @@ bootstrap_post_import_hook ()
     if ! test -f $f; then
       echo "required file $f is missing" 2>&1
       exit 1
     fi
   done
 
   echo "generating source lists for liboctave/operators/module.mk..."
 
-  (cd liboctave/operators; ./config-ops.sh)
+  rm -f liboctave/operators/*-op-*.mk
+  (cd liboctave/operators && ${SHELL:-/bin/sh} config-ops.sh)
 
   echo "generating doc/interpreter/images.mk..."
 
-  (cd doc/interpreter; ./config-images.sh)
+  rm -f doc/interpreter/images.mk
+  (cd doc/interpreter && ${SHELL:-/bin/sh} config-images.sh)
 
   echo "generating libinterp/dldfcn/module.mk..."
 
-  (cd libinterp/dldfcn; ./config-module.sh)
+  rm -f libinterp/dldfcn/module.mk
+  (cd libinterp/dldfcn && ${SHELL:-/bin/sh} config-module.sh)
 }
 
